docs/README.md

```diff
@@ -0,0 +1,178 @@
+
+  This field specifies ECU id and that should be unique in all EUCs.
+
+- ip_addr (string, optional)
+
+  This field specifies this ota client's IP address.  
+  If this field is not specified, "localhost" is used.  
+  NOTE: this IP address is used for the local server bind address.
+
+- secondaries (array, optional)
+
+  This field specifies list of **directly connected** secondary ECUs information.  
+  If this field is not specified, it is treated as if secondary ecu doesn't exist.
+
+  - ecu_id (string, required)
+    ecu id of secondary ECU
+
+  - ip_addr (string, required)
+    IP address of secondary ECU.
+
+- available_ecu_ids (optional)
+
+  This field specifies a list of all ECU ids, including directly connected, indirectly connected and itself.
+  Only the main ECU should have this information.  
+  If this field is not specified, value of `ecu_id` is used as this value.  
+
+  NOTE: The difference between secondaries and available_ecu_ids:
+
+  `secondaries` lists the directly connected children ECUs, available_ecu_ids consists of all children ECU ids(including directly connected, indirectly connected and itself).
+
+##### The default setting
+
+If ecu_info.yml doesn't exist, the default setting is used as follows:
+
+- format_version
+  - 1
+
+- ecu_id
+  - "autoware"
+
+#### proxy\_info.yml
+
+proxy_info.yml is the setting file for OTA proxy configuration.
+
+OTA proxy is the software integrated into the ota client that access the ota server on behalf of the ota client.
+Whether ota proxy access the ota server directly or indirectly depends on the configuration.
+
+See [OTA proxy](../ota_proxy/README.md) more details.
+
+##### File path
+
+/boot/ota/proxy_info.yml
+
+##### Entries
+
+- enable_local_ota_proxy (boolean, optional)
+
+  This field specifies whether ota client uses local ota proxy or not.
+  If this field is not specified, ota client doesn't use local ota proxy, it means ota client connects to ota server directly. If the local ota proxy is enabled, the ota client requests the local ota proxy.
+
+The configuration for local ota proxy are as follows.
+
+- gateway (boolean, optional if `enable_local_ota_proxy` is true otherwise not required)
+
+  When the `enable_local_ota_proxy` field is true, this field specifies whether the **local ota proxy** requests the ota server directly with HTTPS or HTTP. If it is true, HTTPS is used otherwise HTTP is used.  
+  If this field is not specified, HTTP is used.  
+  Note that if the ECU can't access to the ota server directly, the value should be set to false.
+
+- upper_ota_proxy (string, optional if `enable_local_ota_proxy` is true otherwise not required)
+
+  When the `enable_local_ota_proxy` field is true, this field specifies the upper ota proxy address to be accessed by the local ota proxy.  
+  To specify the upper ota proxy address, `http://192.168.20.11:8082` notation is used.
+  If this field is not specified, the **local ota proxy** requests ota server directly.
```

Comment:
@obi-t4 When ota-proxy is not enabled, the ota-client will use this proxy to connect the remote ota proxy. In other word, this option will always be used if it is set.

More details as follow, when `upper_ota_proxy` is set:
1. if local ota-proxy enabled, ota-client will connect to local ota-proxy for requesting files, and `upper_ota_proxy` is used by local ota-proxy.
2. if local ota-proxy disabled, ota-client will connect to `upper_ota_proxy` for requesting files.

Only when `upper_ota_proxy` is not set and `enable_ota_proxy` is `false`, ota-client connects to the remote ota file server directly.

Reply:
`if local ota-proxy disabled, ota-client will connect to upper_ota_proxy for requesting files.` 
Ah, I see. This is what I was misunderstainding.

Reply:
fixed at https://github.com/tier4/ota-client/pull/129/commits/7b7cb44c82569717d0038bebeb81b9604bad4554
![image](https://user-images.githubusercontent.com/58019445/170439233-18fff0eb-f59c-4373-8338-f1304f431f66.png)